JBPM-8044 & JBPM-8037: [Stunner] Docking issues when preview panel is opened

romartin opened the pull request:
Hey @hasys  @tiagodolphine 

Here are the fixes for a couple of bugs reported:
* [JBPM-8044](https://issues.jboss.org/browse/JBPM-8044): [Stunner] Docking is not working when the preview panel is opened
* [JBPM-8037](https://issues.jboss.org/browse/JBPM-8037): [Stunner] Deleting boundary event from node throws GWT error

If you agree on this fix, as it's not risky at all, probably we can even consider adding this into the next CR2 as well, let's keep in touch.

It depends on:
https://github.com/kiegroup/lienzo-core/pull/78
https://github.com/kiegroup/lienzo-tests/pull/56

Thanks!

romartin commented:
Jenkins execute full downstream build

romartin commented:
rebased

romartin commented:
Jenkins execute full downstream build

romartin commented:
Jenkins please retest this

romartin commented:
Please @hasys  @tiagodolphine , this one is a blocker and next week I'll be on PTO, so I would like to leave in peace with it... can you please give priority to this one?? I don't want last time worries... :) 
Thanks in advance!

tiagodolphine commented on kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-lienzo/src/main/java/org/kie/workbench/common/stunner/client/lienzo/canvas/wires/WiresLayer.java:
```diff
@@ -81,7 +81,6 @@ public WiresLayer dock(final WiresContainer parent,
                            final WiresShape child) {
         final WiresDockingControl dockingControl = child.getControl().getDockingControl();
         dockingControl.dock(parent);
-        child.setLocation(dockingControl.getCandidateLocation());
```
Is it necessary to remove this setLocation on the child? I think this may cause a regression when docking directly from the palette.

romartin replied:
Yeah @tiagodolphine  I know we're doing this here but this was causing issues on the preview, also I'm pretty sure is no longer necessary. Consider than from the preview  widget, when a dock command is executed, the `dockingControl` has no real "state", as the user is really not "dragging" on the preview panel, so the `dockingControl.getCandidateLocation()` call doesn't make sense.
Also when the user docks from the UI, in fact 2 commands are executed - dock & update location, so the `setLocation` should be anyway being called. I've tested this and it seems to work fine. WDYT?

romartin replied:
in fact the issue you mention about dragging directly from the palette is due to this line being removed. Anyway I think it does not makes sense and we should split dock  (relationship) and update location (attributes) into different commands / actions. Well for now trying to see if I can fix the issue... will keep you updated. thx man

romartin commented:
Thanks for the tests @tiagodolphine , good point, I forgot to test directly from the palette. Working on it! :+1:

romartin commented:
Jenkins execute full downstream build

tiagodolphine commented:
Jenkins execute full downstream build

romartin commented:
Hey @hasys  @tiagodolphine 

Artifacts generated! This commits will be cherry-picked to the product branch, so please would appreciate if you can review it ASAP in order to create the PRs on the 7.14.x branch before leaving for holidays.

Thanks!

tiagodolphine commented:
Hi @romartin I tested but I still got some issues when docking from the palette, take a look on the two videos https://issues.jboss.org/secure/attachment/12444943/New-Docking.webm https://issues.jboss.org/secure/attachment/12444942/Old-Docking.webm
If you see the docking location is not right, I mean it is not being moved to the parent node border.
Don't know what is the quick solution now, but one option is to disable the docking from pallete for now in case the solution take time. WDYT ?

romartin commented:
Hey @tiagodolphine  @hasys 

Thanks for the review! I've fixed the issues when dropping intermediate events into nodes (tasks), directly from the palette. It works fine for me now, please can you check again? Also just rebased.

The idea is to integrate those fixes about docking into 7.14.x, targeting next 7.2.x release. So please guys can you give some priority to this PR, so I can create the one for 7.14.x ASAP? 

@lazarotti @krisv @ibek please can you confirm if I can go ahead and create the `RHPAM` ticket and integrate those tickets into next product release? Thx!
 
Thanks in advance!

romartin commented:
Jenkins execute full downstream build

romartin commented:
thanks @tiagodolphine ! 

Hey @hasys , could you please give priority to this PR? This is because I would like to integrate these fixes also into 7.14.x, as for next product release. So once I have his merged and verified, I can start cherry-picking it... thanks in advance! :+1:

romartin commented:
@hasys fixed lienzo side :+1: 
thanks!

romartin commented:
Jenkins execute full downstream build

romartin commented:
Jenkins please retest this

romartin commented:
hey @hasys please remember to check this, it's urgent as it's targeted for next 7.2.1 release. Thanks!

hasys commented:
Hi @romartin,

fixes looks good, but unfortunately, I found one small regression (for master branch we can move it to separate to another PR, but for 7.14 cherry-picking I think it's better to provide single PR with fixes).

When you attach catching event to a node, on save for some condition all nodes disappear, and you need to mouse move on any node or click on the canvas to see all nodes back.

#### Steps to reproduce:
1. attach any catch event to any task (not necessary to open Preview panel)
2. deactivate catch event (I am just clicking on the canvas on empty space)
3. save process

#### Actual Result:
All content disappear, you need to move your mouse over any node or click on the canvas to see Processes content.
#### Expected Result:
Process will be saved, content will be displayed without any issues.

tiagodolphine commented:
Hi @hasys I think this issue was fixed here https://github.com/kiegroup/kie-wb-common/pull/2378, maybe if we run the full downstream again, it should be fixed since the PR was merged yesterday.

hasys commented:
Hi @tiagodolphine,

ohh, probably that's why I can't reproduce it on master, thank you @tiagodolphine!

Let's try full downstream build, should be done today so let's try it.

Jenkins execute full downstream build

romartin commented:
yeah @hasys that bug is already fixed... thanks @tiagodolphine! :+1:

romartin commented:
BTW @hasys the bug you mention about shapes disappearing, which @tiagodolphine already fixed, only applied to master anyway, as it was introduced as a regression for the infinite canvas thing, so it should NOT be present on the `7.14.x` branch.
